src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -117,9 +120,10 @@ internal partial class Binder
             }
 
             ConstantValue constantValue = this.FoldConstantConversion(syntax, source, conversion, destination, diagnostics);
-            if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.Kind == BoundKind.DefaultExpression)
+            if (conversion.Kind == ConversionKind.DefaultOrNullLiteral && source.KindIgnoringSuppressions() == BoundKind.DefaultExpression)
             {
-                source = ((BoundDefaultExpression)source).Update(constantValue, destination);
+                var result = ((BoundDefaultExpression)source.RemoveSuppressions()).Update(constantValue, destination);
+                source = result.WrapWithSuppressionsFrom(source);
```

Comment:
I realize you're looking for higher-level feedback, but could these cases be written with a special `Update` method that handles unwrapping and re-wrapping?
```C#
source = UpdateWithSuppressions(e =>
    ((BoundDefaultExpression)e).Update(constantValue, destination);
```